File: frontend/src/services/authService.js
/**
 * Auth Service
 */
import apiClient, { setTokens, clearTokens } from '../lib/apiClient';

export const authService = {
    register: async ({ name, email, password }) => {
        const res = await apiClient.post('/auth/register', { name, email, password });
        setTokens(res.data.access_token, res.data.refresh_token);
        return res.data;
    },

    login: async ({ email, password }) => {
        const res = await apiClient.post('/auth/login', { email, password });
        setTokens(res.data.access_token, res.data.refresh_token);
        return res.data;
    },

    logout: async () => {
        try { await apiClient.post('/auth/logout'); } catch { }
        clearTokens();
    }
};

export default authService;

File: backend/app/services/auth_service.py
"""Auth Service"""
from flask_jwt_extended import create_access_token, create_refresh_token
from app.extensions import db
from app.models import User

class AuthError(Exception):
    def __init__(self, msg, code=400):
        self.message = msg
        self.status_code = code

def register(name, email, password):
    if not all([name, email, password]):
        raise AuthError('All fields required', 400)
    if User.query.filter_by(email=email).first():
        raise AuthError('Email exists', 409)
    
    user = User(name=name, email=email)
    user.set_password(password)
    db.session.add(user)
    db.session.commit()
    
    return {
        'user': user.to_dict(),
        'access_token': create_access_token(identity=str(user.id)),
        'refresh_token': create_refresh_token(identity=str(user.id))
    }

def login(email, password):
    if not email or not password:
        raise AuthError('Email and password required', 400)
    
    user = User.query.filter_by(email=email).first()
    if not user or not user.check_password(password):
        raise AuthError('Invalid credentials', 401)
    
    return {
        'user': user.to_dict(),
        'access_token': create_access_token(identity=str(user.id)),
        'refresh_token': create_refresh_token(identity=str(user.id))
    }

def refresh(user_id):
    return create_access_token(identity=str(user_id))

def get_user(user_id):
    return User.query.get(user_id)
